src/reactor.rs

```diff
@@ -0,0 +1,106 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! This module is about different kinds of block and event reactors.
+
+use auxiliary;
+use ethereum::types::block::Block;
+use ethereum::Ethereum;
+use std::sync::Arc;
+use web3::contract::Contract;
+use web3::transports::Http;
+use web3::types::Address;
+
+use Config;
+
+/// This enum represents all reactors which will react to block generation.
+pub enum Reactor {
+    BlockReporter {
+        block_store: Arc<Contract<Http>>,
+        validator_address: Address,
+    },
+}
+
+/// Anything that wants to react on block generation should implement this.
+pub trait React {
+    /// Defines how different reactor will react on block observation.
+    ///cargo
+    /// # Arguments
+    ///
+    /// * `block` - The observed block.
+    /// * `event_loop` - The reactor's event loop to handle the tasks spawned.
+    fn react(&self, block: &Block, event_loop: &tokio_core::reactor::Handle);
+}
+
+impl React for Reactor {
```

Comment:
Do we need the trait? Will some other object implement the trait as well? So far it is only implemented by the `Reactor` enum. `react` could be added to `impl Reactor {`.